### ISSUE TITLE
Geolocation information is not exposed in JSON

### ISSUE THREAD

**Issue by SCdF** [medic/medic-webapp#2450]
Currently (nearly) all data from XML forms is extracted out into the JSON. The only data that I've noticed that is missing AND actually useful is the GPS lat and long data.

Work out:
 - Why our extraction code is not extracting geo data
 - If it's intentional
 - Fix it if it's not

At some point we'll need to write some medic-analytics code to extract just this value out of the XML into a legacy extraction table to make up the difference (since we're not going to go back and update existing docs), but that can wait until closer to when someone actually wants to use that data for something.

**Comment by SCdF** [comment]
So some geo data from historical versions is available at `fields/geolocation`, but the majority of documents are missing JSON geo data because is it inside a `meta` tag in the XML, and we are intentionally skipping meta tags.

**Comment by SCdF** [comment]
The extraction of XML into `fields` was one in #1121, including explicitly ignoring meta. There are no code comments, git comments, or ticket comments to indicate why that decision was made.

My vote would be to remove this restriction.

**Comment by SCdF** [comment]
@alxndrsn and @garethbowen are you guys cool with this PR: https://github.com/medic/medic-webapp/pull/2451

Linking to both of you to increase the chance that someone will know background info around this.

**Comment by garethbowen** [comment]
Before couch2pg existed we used `fields` for what to display when viewing a report. `meta` was removed because we don't want to show meta information to users (it's mostly just clutter). So this PR will negatively impact the report content view.

I recommend adding a little code to the `getDisplayFields` function of the Reports controller to ignore the meta information so it's still available for analytics but hidden in the view.

You could argue that the meta information is useful in the view but the raw lat and long values almost certainly aren't.

**Comment by SCdF** [comment]
@garethbowen so on testing this `meta` is actually hidden from the view anyway, because it's inside `input`, which is currently hidden.

**Comment by alxndrsn** [comment]
Seems weird that `meta` is inside `input`...

**Comment by SCdF** [comment]
So it makes sense (sort of) that geolocation is inside `input`, since it's an input from the android wrapper. Why it's also inside `meta`, or why `meta` is inside `input`, is another question yeah

**Comment by alxndrsn** [comment]
Exactly - I'd expect it to be in `/meta` or `/inputs` (or possibly both..!)  But not in `/inputs/meta`.

**Comment by SCdF** [comment]
Okay looking at this more, there are two `meta` tags. Here is an example of a fields structure with `meta` exposed and some data removed:
```
   "fields": {
       "inputs": {
           "meta": {
               "location": {
                   "lat": "",
                   "long": ""
               },
               "deprecatedID": ""
           },
           "source": "task",
           "source_id": "",
           "contact": {
               "_id": "9278D019-AD79-B2A2-A2D8-A8E22E45D6DA",
               "name": "Test Test Family"
           }
       },
       "place_id": "9278D019-AD79-B2A2-A2D8-A8E22E45D6DA",
...
       "meta": {
           "instanceID": "uuid:89538081-e65d-4bba-9cda-ed5453951ce0"
       }
   }
```

So I guess hiding `meta` makes sense as well. It's pretty terrible it's in two places at once though.

**Comment by SCdF** [comment]
So @garethbowen / @alxndrsn what are your thoughts on what I posted above? 

My questions are:
 - Is it OK to just dump everything (including both metas) out of the XML into JSON and then deal with the UI by adding them in the hide list?
 - Is it OK / correct that the XML has two meta tags?
 - Does it really matter where location is? Should it perhaps be moved to somewhere else in the XML, perhaps outside of meta? How would that gel with existing documents?

My (naive) vote would be to just expose everything in the JSON even though it's a bit weird.

**Comment by alxndrsn** [comment]
a) i don't see a problem with this
2. I don't think `/inputs/meta` should exist; perhaps there is a reason for it?
c. `/meta/location` may be auto-filled on openrosa, and maintaining compatibility with that seems reasonable.  I will check what they do.

**Comment by garethbowen** [comment]
I have nothing to add.

**Comment by SCdF** [comment]
Merged.

**Comment by abbyad** [comment]
Is this available somewhere to be acceptance tested? @michaelkohn is waiting on this fix for a project.

Is this being included in the upcoming v2.8.0 release?

**Comment by sglangevin** [comment]
I've acceptance tested this on lg.dev (v2.9.0-beta.4) and it's working well. Moving to ready.
